declarations/Module.d.ts

```diff
@@ -0,0 +1,337 @@
+    /**
+     * @abstract
+     * @returns {Set<string>} types availiable (do not mutate)
+     */
+    getSourceTypes(): Set<string>;
+    /**
+     * @abstract
+     * @deprecated Use codeGeneration() instead
+     * @param {SourceContext} sourceContext source context
+     * @returns {Source} generated source
+     */
+    source(sourceContext: SourceContext): import("webpack-sources").Source;
+    /**
+     * @abstract
+     * @param {string=} type the source type for which the size should be estimated
+     * @returns {number} the estimated size of the module (must be non-zero)
+     */
+    size(type?: string): number;
+    /**
+     * @param {LibIdentOptions} options options
+     * @returns {string | null} an identifier for library inclusion
+     */
+    libIdent(options: LibIdentOptions): string;
+    /**
+     * @returns {string | null} absolute path which should be used for condition matching (usually the resource path)
+     */
+    nameForCondition(): string;
+    /**
+     * @abstract
+     * @deprecated Use codeGeneration() instead
+     * Get a list of runtime requirements
+     * @param {SourceContext} context context for code generation
+     * @returns {Iterable<string> | null} required runtime modules
+     */
+    getRuntimeRequirements(context: SourceContext): Iterable<string>;
+    /**
+     * @param {CodeGenerationContext} context context for code generation
+     * @returns {CodeGenerationResult} result
+     */
+    codeGeneration(context: CodeGenerationContext): CodeGenerationResult;
+    /**
+     * @param {Chunk} chunk the chunk which condition should be checked
+     * @param {Compilation} compilation the compilation
+     * @returns {boolean} true, if the chunk is ok for the module
+     */
+    chunkCondition(chunk: import("./Chunk"), compilation: import("./Compilation")): boolean;
+    /**
+     * Assuming this module is in the cache. Update the (cached) module with
+     * the fresh module from the factory. Usually updates internal references
+     * and properties.
+     * @param {Module} module fresh module
+     * @returns {void}
+     */
+    updateCacheModule(module: Module): void;
+    /**
+     * @returns {Source | null} the original source for the module before webpack transformation
+     */
+    originalSource(): import("webpack-sources").Source;
+    serialize(context: any): void;
+    deserialize(context: any): void;
+    useSourceMap: any;
+    get hasEqualsChunks(): any;
+    get isUsed(): any;
+    set used(arg: any);
+    get used(): any;
+}
+declare namespace Module {
+    export { Source, Chunk, ChunkGroup, Compilation, Dependency, DependencyTemplates, FileSystemInfo, UsageStateType, RequestShortener, RuntimeTemplate, WebpackError, Hash, SortableSet, SourceContext, CodeGenerationContext, CodeGenerationResult, LibIdentOptions, KnownBuildMeta, NeedBuildContext, BuildMeta, OptimizationBailoutFunction };
```

Comment:
Semantic-wise this is correct, since `@typedef` reexports the types. Not sure if I like it that much. Would be nice if this would be configurable. `noTypedefExport`?